sw/device/tests/sensor_ctrl_wakeup.c

```diff
@@ -0,0 +1,134 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_sensor_ctrl.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+static dif_pwrmgr_t pwrmgr;
+static dif_rv_plic_t plic;
+static volatile bool correct_irq;
+
+/**
+ * External interrupt handler.
+ */
+void ottf_external_isr(void) {
+  correct_irq = false;
+  dif_rv_plic_irq_id_t irq_id;
+  top_earlgrey_plic_peripheral_t peripheral;
+
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_claim(&plic, kTopEarlgreyPlicTargetIbex0, &irq_id));
+
+  peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[irq_id];
+
+  dif_pwrmgr_irq_t irq =
+      (dif_pwrmgr_irq_t)(irq_id - kTopEarlgreyPlicIrqIdPwrmgrAonWakeup);
+
+  // Check that both the peripheral and the irq id is correct
+  if (peripheral == kTopEarlgreyPlicPeripheralPwrmgrAon) {
+    if (irq_id == kTopEarlgreyPlicIrqIdPwrmgrAonWakeup) {
+      CHECK_DIF_OK(dif_pwrmgr_irq_acknowledge(&pwrmgr, irq));
+      correct_irq = true;
+    }
+  }
+
+  // Complete the IRQ by writing the IRQ source to the Ibex specific CC.
+  // register
+  CHECK_DIF_OK(
+      dif_rv_plic_irq_complete(&plic, kTopEarlgreyPlicTargetIbex0, irq_id));
+}
```

Comment:
Would this help you: https://github.com/lowRISC/opentitan/blob/6c31d3c9048cd84a77539549af3889cda251536d/sw/device/lib/testing/autogen/isr_testutils.c#L569 ?

Reply:
updated! this definitely helps